category_theory/discrete_category.lean

```diff
@@ -0,0 +1,77 @@
+-- Copyright (c) 2017 Scott Morrison. All rights reserved.
+-- Released under Apache 2.0 license as described in the file LICENSE.
+-- Authors: Stephen Morgan, Scott Morrison
+
+import data.ulift
+import category_theory.natural_transformation
+import category_theory.isomorphism
+import category_theory.functor_category
+
+namespace category_theory
+
+universes u₁ v₁ u₂ v₂
+
+def discrete (α : Type u₁) := α
+
+@[extensionality] lemma plift.ext {P : Prop} (a b : plift P) : a = b :=
+begin
+  cases a, cases b, refl
+end
+
+instance plift_subsingleton (P : Prop) : subsingleton (plift P) :=
```

Comment:
Similar comments also apply to orphaning instances. In fact this instance already exists, in `data.equiv.basic`; evidently also not the best place for it!

Reply:
I've removed this instance from `discrete_category.lean`, and not immediately planning on rearranging the existing instances.